src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -223,6 +235,40 @@ public function addHtmlContent($content, $charset = 'UTF-8')
         }
     }
 
+    /**
+     * @param string $content The HTML content
+     *
+     * @return \DOMDocument
+     */
+    private function parseHtml5($content)
```

Comment:
is the part about `$content = mb_convert_encoding($content, 'HTML-ENTITIES', $charset);` necessary for the html5 loading ? If no, it should be moved inside your new `parseXhtml` method